dev/io.openliberty.checkpoint_fat/fat/src/io/openliberty/checkpoint/fat/MPConfigTest.java

```diff
@@ -52,68 +62,117 @@ public static void copyAppToDropins() throws Exception {
 
     @Before
     public void setUp() throws Exception {
-        TestMethod testMethod = getTestMethod(TestMethod.class, testName);
-        configureBeforeCheckpoint(testMethod);
+        testMethod = getTestMethod(TestMethod.class, testName);
+        configureBeforeCheckpoint();
         server.setCheckpoint(CheckpointPhase.APPLICATIONS, true,
                              server -> {
                                  assertNotNull("'SRVE0169I: Loading Web Module: " + APP_NAME + "' message not found in log before rerstore",
                                                server.waitForStringInLogUsingMark("SRVE0169I: .*" + APP_NAME, 0));
                                  assertNotNull("'CWWKZ0001I: Application " + APP_NAME + " started' message not found in log.",
                                                server.waitForStringInLogUsingMark("CWWKZ0001I: .*" + APP_NAME, 0));
-                                 configureBeforeRestore(testMethod);
+                                 configureBeforeRestore();
                              });
         server.startServer(getTestMethod(TestMethod.class, testName) + ".log");
     }
 
-    private void configureBeforeCheckpoint(TestMethod testMethod) throws Exception {
-        if (testMethod == TestMethod.envValueChangeTest || testMethod == TestMethod.appScopeEnvValueChangeTest) {
-            try {
-                server.copyFileToLibertyServerRoot("envValueTest/server.env");
-            } catch (Exception e) {
-                throw new AssertionError("Unexpected error configuring test.", e);
-            }
+    private void configureBeforeCheckpoint() throws Exception {
+        switch (testMethod) {
+            case envValueChangeTest:
+                configureEnvVariable("req_scope_key", "envValue");
+                break;
+            case appScopeEnvValueChangeTest:
+                configureEnvVariable("app_scope_key", "envValue");
+                break;
+            case configObjectAppScopeEnvValueChangeTest:
+                configureEnvVariable("config_object_app_scope_key", "envValue");
+                break;
+            case configObjectPropertiesAppScopeEnvValueChangeTest:
+                configureEnvVariable("config_object_properties_app_scope_key", "envValue");
+                break;
+            default:
+                break;
+        }
+    }
```

Comment:
This confuses me. You have several tests with "change" in the name where the environment variable is set to one value before checkpoint and a different value afterwards.

However, for all the other tests, you're not setting the value before checkpoint, you're only setting it afterwards. I.e. the value still changes from unset to set.

You don't appear to be doing any tests where the value is the same before checkpoint and on restore?

Reply:
> This confuses me. You have several tests with "change" in the name where the environment variable is set to one value before checkpoint and a different value afterwards.
> 
> However, for all the other tests, you're not setting the value before checkpoint, you're only setting it afterwards. I.e. the value still changes from unset to set.

From what I understand of the test all keys used should have some default value in the server.xml that is changed before restore by using a config source that will override the `defaultValue` from the server.xml.

> 
> You don't appear to be doing any tests where the value is the same before checkpoint and on restore?

We can certainly add something like that to make sure things work when you don't change the value.

Reply:
> We can certainly add something like that to make sure things work when you don't change the value.

Actually we already have that with the tests:

 defaultValueTest:
 appScopeDefaultValueTest:
 configObjectAppScopeDefaultValueTest:
 configObjectPropertiesAppScopeDefaultValueTest:

Reply:
Thanks. After being confused by this, I decided I needed to write out the whole test matrix to see exactly what was an wasn't tested. See my new comments.